core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -96,7 +96,7 @@ void HIP::impl_initialize(InitializationSettings const& settings) {
   // Allocate a staging buffer for constant mem in pinned host memory
   // and an event to avoid overwriting driver for previous kernel launches
   KOKKOS_IMPL_HIP_SAFE_CALL(
-      hipHostMalloc((void**)&Impl::HIPInternal::constantMemHostStaging,
+      hipHostMalloc(reinterpret_cast<void**>(&Impl::HIPInternal::constantMemHostStaging),
                     Impl::HIPTraits::ConstantMemoryUsage));
```

Comment:
Can I invoke memory safety concerns (https://www.c-faq.com/ptrs/genericpp.html) to replace this `reinterpret_cast` with something like
```cpp
  void* void_ptr = nullptr;
  KOKKOS_IMPL_HIP_SAFE_CALL(hipHostMalloc(&void_ptr, Impl::HIPTraits::ConstantMemoryUsage));
  Impl::HIPInternal::constantMemHostStaging = void_ptr;
```

Reply:
I agree that this is something we should consider fixing.
Considering that I was able to find a few more instances of this pattern with a quick search, I am fine with us doing it as a follow up.